osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -289,7 +293,32 @@ bool attemptSelection(CarouselItem item)
         protected GroupedBeatmapSet? ExpandedBeatmapSet { get; private set; }
 
         protected override bool ShouldActivateOnKeyboardSelection(CarouselItem item) =>
-            grouping.BeatmapSetsGroupedTogether && item.Model is BeatmapInfo;
+            grouping.BeatmapSetsGroupedTogether && item.Model is GroupedBeatmap;
+
+        public override object? CurrentSelection
```

Comment:
As you mentioned in the commit messaging, this is controversial. But I'd only say it is by your own original requirements for the carousel which I've been sticking to stringently (which is that `CurrentSelection` is never mutated internally).

Which is to say, I don't have much issue with this flow, as long as you're okay with it existing.

The only real alternative I can think of is hiding (to the best of our ability, either going the full `protected` path to avoid generics infesting the carousel even further, or just `new` in `BeatmapCarousel`) `object? CurrentSelection` and exposing two typed setters for `GroupedBeatmap` and `BeatmapInfo`, to offer more explicit pathways to getting the correct model types in.